File: sistema-escola/src/models/Pessoa.js
/**
 * Classe Abstrata: Pessoa
 * Não pode ser instanciada diretamente.
 * Atributos privados: #nome, #cpf
 */
class Pessoa {
  #nome;
  #cpf;

  constructor(nome, cpf) {
    if (new.target === Pessoa) {
      throw new Error("Pessoa é uma classe abstrata e não pode ser instanciada diretamente.");
    }
    this.#nome = nome;
    this.#cpf = cpf;
  }

  getNome() { return this.#nome; }
  setNome(nome) { this.#nome = nome; }

  getCpf() { return this.#cpf; }
  setCpf(cpf) { this.#cpf = cpf; }

  // Método abstrato — deve ser sobrescrito pelas subclasses
  mostrarDados() {
    throw new Error("mostrarDados() deve ser implementado pela subclasse.");
  }

  // Método da interface Autenticavel — deve ser sobrescrito
  login() {
    throw new Error("login() deve ser implementado pela subclasse.");
  }
}

export default Pessoa;

File: src/main/java/escola/Pessoa.java
package escola;

/**
 * Classe Abstrata: Pessoa
 * Não pode ser instanciada diretamente.
 * Atributos privados: nome e cpf.
 */
public abstract class Pessoa {

    private String nome;
    private String cpf;

    public Pessoa(String nome, String cpf) {
        this.nome = nome;
        this.cpf  = cpf;
    }

    // Getters e Setters
    public String getNome() { return nome; }
    public void   setNome(String nome) { this.nome = nome; }

    public String getCpf() { return cpf; }
    public void   setCpf(String cpf) { this.cpf = cpf; }

    // Método abstrato — obrigatório nas subclasses
    public abstract void mostrarDados();
}
